sdk/src/wallet/account/operations/transaction/high_level/send_nft.rs

```diff
@@ -1,26 +1,41 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-// use primitive_types::U256;
+use getset::Getters;
 use serde::{Deserialize, Serialize};
 
 use crate::{
     client::api::PreparedTransactionData,
     types::block::{
-        address::Bech32Address,
+        address::{Bech32Address, Bech32AddressLike},
         output::{unlock_condition::AddressUnlockCondition, NftId, NftOutputBuilder, Output},
     },
     wallet::account::{operations::transaction::Transaction, Account, TransactionOptions},
 };
 
 /// Params for `send_nft()`
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize, Getters)]
 #[serde(rename_all = "camelCase")]
 pub struct SendNftParams {
     /// Bech32 encoded address
-    pub address: Bech32Address,
+    #[getset(get = "pub")]
+    address: Bech32Address,
     /// Nft id
-    pub nft_id: NftId,
+    #[getset(get = "pub")]
+    nft_id: NftId,
+}
+
+impl SendNftParams {
+    /// Creates a  new instance of [`SendNftParams`]
+    pub fn new(
+        address: impl Bech32AddressLike,
+        nft_id: impl TryInto<NftId, Error = impl Into<crate::wallet::Error>>,
```

Comment:
Should we also have a (generic?)`*Like` trait for IDs. Seems a bit inconsistent to me that we use both conversion traits here? Probably a separate issue tho.

Reply:
Yes, I was thinking about it yesterday. Additionally, technically it would be possible to eliminate the implementation for `Infalliable` here https://github.com/qrayven/iota-sdk/blob/aaa7e49559e9e2ee7384a41cfe833c40a2f18515/sdk/src/wallet/error.rs#L140, mentioned by @Alexandcoats. But, yeah, maybe a separate issue for that?

Reply:
Yeah, that `From<Infallible>` is really weird. Would be nice to get rid of it.

Reply:
On the other hand, we can't implement a trait for every type we have. At some point the code load outweighs the benefit.